src/components/home/Popup.tsx

```diff
@@ -0,0 +1,184 @@
+/** @jsx jsx */
+import React, {useEffect} from 'react';
+import { RiCloseCircleLine } from 'react-icons/ri';
+import { jsx, SxStyleProp } from 'theme-ui';
+
+import { fadeIn } from '../../utils/animation';
+import { IPost } from '../../utils/interface';
+
+export interface PopupProps {
+  /** The post details. */
+  post: IPost;
+  /** The callable that is called once the close button is pressed. */
+  closeHandler: () => void;
+}
+
+/**
+ * A component representing a popup for a post.
+ * 
+ * This popup will occur once a post is clicked, and contains the full
+ * message, the author's name, and the picture in its original
+ * aspect ratio.
+ * 
+ * A popup can only be for posts that have an image. If a popup is formed
+ * from a post that has no image, no popup will show.
+ * 
+ * Clicking the X button or the borders of the popup will close the popup.
+ * 
+ * Desktop view:
+ *  - Spans the entire page
+ *  - The image area is about 60% of the width, and the text area is the rest
+ *  - A small header above the image area contains the exit button
+ * 
+ * Mobile view:
+ *  - Only the picture will be shown in mobile view, as the text can be
+ *    easily found by clicking the description.
+ * 
+ * @returns the popup of a post.
+ */
+export const Popup: React.FC<PopupProps> = ({post, closeHandler}) => {
+  const {name, message, photo} = post;
+
+  // Prevent the body from scrolling in a popup
+  useEffect(() => {
+    // Haha, get it? Old overflow = oldverflow?
+    const oldverflow = document.body.style.overflow
+    document.body.style.overflow = "hidden"
+    
+    return () => {
+      document.body.style.overflow = oldverflow;
+    };
+  }, [])
+
+  const popupDiv: SxStyleProp = {
+    position: 'fixed',
+    margin: 0,
+    padding: 0,
+    display: 'flex',
+    flexDirection: 'row',
+    backgroundColor: 'popup.popupBg',
+    top: 0,
+    left: 0,
+    zIndex: 1000,
+
+    '@keyframes fadeIn': fadeIn,
+    animationName: 'fadeIn',
+    animationDuration: '0.25s',
+  }
+
+  const photoRegionDiv: SxStyleProp = {
+    height: '100vh',
+    display: 'flex',
+    flexDirection: 'column',
+    width: ['100vw', '65vw'],
+    ml: 0,
+    mr: 'auto',
+  }
+
+  const textRegionDiv: SxStyleProp = {
+    height: '100vh',
+    width: '35vw',
+    mr: 0,
+    ml: 'auto',
+    px: '2%',
+    py: '2%',
+    backgroundColor: 'popup.textBg',
+    visibility: ['collapse', 'visible'],
+  }
+
+  const authorTextStyle: SxStyleProp = {
+    fontSize: 'medium',
+    fontFamily: 'body',
+  }
+
+  const lineStyle: SxStyleProp = {
+    border: '1px solid gray',
+    height: '2px',
+    width: '100%',
+    margin: 'auto',
+    mb: '7%'
+  }
+
+  const textBoxStyle: SxStyleProp = {
+    overflowY: 'auto',
+    width: '100%',    
+    maxHeight: '85%',
+    '&::-webkit-scrollbar': {
+      width: '0.5em'
+    },
+    '&::-webkit-scrollbar-thumb': {
+      backgroundColor: 'rgba(0, 0, 0, 0.05)',
+      outline: '1px solid slategrey'
+    }
```

Comment:
Can you make the scrollbar to not have outlines and perhaps rounded corners

Reply:
It's a temp scroll bar, I wanted to get this push out now and not focus too much on the very little details (issue time?)